src/test/java/guitests/SaveCommandTest.java

```diff
@@ -27,6 +35,54 @@ public void save() {
         TodoList todoListAtSavedLocation = TodoList.load();
         assertTrue(TestUtil.compareTasksOfTodoLists(todoListAtSavedLocation,
                                                     new TypicalTestTodoLists().getTypicalTodoList()));
+        assertResultMessage(String.format(StoreController.RESULT_MESSAGE_SAVING_SUCCESS, newPath));
+    }
+
+    @Test
+    public void save_fileExists() {
+        String newPath = TestUtil.getFilePathInSandboxFolder("existing_file.json");
+        try {
+            FileUtil.createFile(new File(newPath));
+            String command = "save " + newPath;
+            commandBox.runCommand(command);
+
+            // Check that storage path is changed
+            assertEquals(Config.getInstance().getTodoListFilePath(), newPath);
+
+            // Check that todo list loaded is the same as previous todo list
+            TodoList todoListAtSavedLocation = TodoList.load();
+            assertTrue(TestUtil.compareTasksOfTodoLists(todoListAtSavedLocation,
+                    new TypicalTestTodoLists().getTypicalTodoList()));
+            assertResultMessage(String.format(StoreController.RESULT_MESSAGE_WARNING_OVERWRITE, newPath)
+                + "\n" + String.format(StoreController.RESULT_MESSAGE_SAVING_SUCCESS, newPath));
+        } catch (IOException e) {
+            fail();
```

Comment:
As in i think might be good for logging the error message or something in case u ever reach this line.